cmd/root.go

```diff
@@ -216,6 +217,17 @@ var rootCmd = &cobra.Command{
 				os.Exit(1)
 			}
 		}
+		if err = (&generatorstate.Reconciler{
+			Client:     mgr.GetClient(),
+			Log:        ctrl.Log.WithName("controllers").WithName("GeneratorState"),
+			Scheme:     mgr.GetScheme(),
+			RestConfig: mgr.GetConfig(),
+		}).SetupWithManager(mgr, controller.Options{
+			MaxConcurrentReconciles: concurrent,
+		}); err != nil {
+			setupLog.Error(err, errCreateController, "controller", "GeneratorState")
+			os.Exit(1)
+		}
```

Comment:
One thing we discussed in the community meeting - maybe it will help if this was a separate controller to handle generation - leveraging the same in-tree pkg we have.

Reply:
> maybe it will help if this was a separate controller to handle generation

@gusfcarvalho not sure if i'm getting it. You mean a separate controller per Generator (e.g. `Kind=ACRAccessToken`)? What would it do and how would we transfer values between ExternalSecret and the generator?

Reply:
I was just capturing a comment I thought you did during the community meeting lol :P :P

From what I understood it would be having a `cmd/new-controller.go` which would iterate over the new GeneratorState resource and focus on cleanup logic

But maybe I misunderstood lol I was really just trying to capture community meeting notes in here 😅

Reply:
I see, i meant there's _is_ now a dedicated custom resource which stores the state and a associated controller with it which does the cleanup. No need for an additional entrypoint for that controller though :)